Source/Core/DolphinQt/Debugger/ThreadWidget.cpp

```diff
@@ -0,0 +1,481 @@
+    addr = PowerPC::HostIsRAMAddress(addr) ? PowerPC::HostRead_U32(addr) : 0;
+    return get_hex(addr);
+  };
+  const auto get_string_from = [](u32 addr) {
+    if (!PowerPC::HostIsRAMAddress(addr))
+      return QStringLiteral("\"\"");
+    return QStringLiteral("\"%1\"").arg(QString::fromStdString(PowerPC::HostGetString(addr)));
+  };
+  const auto get_state = [](u16 state) {
+    QString state_name;
+    switch (state)
+    {
+    case 1:
+      state_name = tr("READY");
+      break;
+    case 2:
+      state_name = tr("RUNNING");
+      break;
+    case 4:
+      state_name = tr("WAITING");
+      break;
+    case 8:
+      state_name = tr("MORIBUND");
+      break;
+    default:
+      state_name = tr("UNKNOWN");
+    }
+    return QStringLiteral("%1 (%2)").arg(QString::number(state), state_name);
+  };
+  const auto get_priority = [](u16 base, u16 effective) {
+    return QStringLiteral("%1 (%2)").arg(QString::number(base), QString::number(effective));
+  };
+  const auto get_stack = [](u32 end, u32 start) {
+    return QStringLiteral("%1\n%2")
+        .arg(end, 8, 16, QLatin1Char('0'))
+        .arg(start, 8, 16, QLatin1Char('0'));
+  };
+
+  // YAGCD - Section 4.2.1.4 Dolphin OS Globals
+  m_current_context->setText(get_hex_from(0x800000D4));
+  m_current_thread->setText(get_hex_from(0x800000E4));
+  m_default_thread->setText(get_hex_from(0x800000D8));
+
+  m_queue_head->setText(get_hex_from(0x800000DC));
+  m_queue_tail->setText(get_hex_from(0x800000E0));
+
+  // Thread group
+  m_threads = PowerPC::debug_interface.GetThreads();
+  int i = 0;
+  m_thread_table->setRowCount(i);
+  for (const auto& thread : m_threads)
+  {
+    m_thread_table->insertRow(i);
+    m_thread_table->setItem(i, 0, new QTableWidgetItem(get_hex(thread->GetAddress())));
+    m_thread_table->setItem(i, 1, new QTableWidgetItem(get_state(thread->GetState())));
+    m_thread_table->setItem(i, 2, new QTableWidgetItem(QString::number(thread->IsDetached())));
+    m_thread_table->setItem(i, 3, new QTableWidgetItem(QString::number(thread->IsSuspended())));
+    m_thread_table->setItem(i, 4,
+                            new QTableWidgetItem(get_priority(thread->GetBasePriority(),
+                                                              thread->GetEffectivePriority())));
+    m_thread_table->setItem(
+        i, 5, new QTableWidgetItem(get_stack(thread->GetStackEnd(), thread->GetStackStart())));
+    m_thread_table->setItem(i, 6, new QTableWidgetItem(QString::number(thread->GetErrno())));
+    m_thread_table->setItem(i, 7,
+                            new QTableWidgetItem(QString::fromStdString(thread->GetSpecific())));
+    i += 1;
+  }
+  m_thread_table->resizeColumnsToContents();
+  m_thread_table->resizeRowsToContents();
+
+  // Thread's context group
+  const Common::Debug::PartialContext empty_context;
+  UpdateThreadContext(&empty_context);
+}
+
+void ThreadWidget::UpdateThreadContext(const Common::Debug::PartialContext* context)
```

Comment:
I'd pass a const-ref instead of a pointer here. Pointer semantics don't seem useful here.

And then you can just do `UpdateThreadContext({})` in the other functions to reset.

```suggestion
void ThreadWidget::UpdateThreadContext(const Common::Debug::PartialContext& context)
```